File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        final int[] result = new int[2];
        final Map<Integer, Integer> indexMap = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            final int difference = target - nums[i];
            if (indexMap.containsKey(difference)) {
                result[0] = indexMap.get(difference);
                result[1] = i;
                return result;
            }
            indexMap.put(nums[i], i);
        }
        return null;
    } 
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int size = nums.size();
        for (int i=0;i<size-1;i++) {
            for (int j=i+1;j<size;j++) {
                if (nums[j]==nums[i]) return true;
            }
        }
        return false;
    }
};
